### ISSUE TITLE
How the inverse was calculated

### ISSUE THREAD

**Issue by jiaqingxie** [VincentStimper/normalizing-flows#40]
Hi, 
I do not understand how to compute the inverse normalizing flow. Could you explain how do you compute the inverse function? Thank you.
Link:
https://github.com/VincentStimper/normalizing-flows/blob/ce86fe79d7cdc34f0362f487194d0cebb1999edc/normflows/flows/planar.py#LL66C3-L66C3
```python
  def inverse(self, z):
        if self.act != "leaky_relu":
            raise NotImplementedError("This flow has no algebraic inverse.")
        lin = torch.sum(self.w * z, list(range(1, self.w.dim()))) + self.b
        a = (lin < 0) * (
            self.h.negative_slope - 1.0
        ) + 1.0  # absorb leakyReLU slope into u
        inner = torch.sum(self.w * self.u)
        u = self.u + (torch.log(1 + torch.exp(inner)) - 1 - inner) \
            * self.w / torch.sum(self.w ** 2)
        dims = [-1] + (u.dim() - 1) * [1]
        u = a.reshape(*dims) * u
        inner_ = torch.sum(self.w * u, list(range(1, self.w.dim())))
        z_ = z - u * (lin / (1 + inner_)).reshape(*dims)
        log_det = -torch.log(torch.abs(1 + inner_))
        return z_, log_det
```

**Comment by VincentStimper** [comment]
Hi @jiaqingxie,

When the activation function is a leaky ReLU, the planar flow is simple a piecewise linear function and can hence be inverted like a linear map on the two sections.

Best regards,
Vincent
